src/main/java/ch/njol/skript/lang/parser/ParserInstance.java

```diff
@@ -356,6 +360,57 @@ public Kleenean getHasDelayBefore() {
 		return hasDelayBefore;
 	}
 
+	// Return API
+
+	private Deque<ReturnData> returnStack = new LinkedList<>();
+
+	public Deque<ReturnData> getReturnStack() {
+		return returnStack;
+	}
+
+	public void setReturnStack(Deque<ReturnData> returnStack) {
+		this.returnStack = returnStack;
+	}
+
+	/**
+	 * Retrieves the current {@link ReturnData}
+	 * @return the return data
+	 */
+	@Nullable
+	public ReturnData getCurrentReturnData() {
+		return returnStack.peek();
+	}
```

Comment:
I think this should be an Optional around the nullable, because most the time it won't exist, but it may so an Optional could make sense here.